sdk/java/pulumi/src/main/java/com/pulumi/resources/StackReference.java

```diff
@@ -221,6 +223,29 @@ public Output<?> requireOutput(Output<String> name) {
         return Internal.of(value).withIsSecret(isSecretOutputName(name));
     }
 
+    /**
+     * Fetches the value of the named stack output
+     * and builds a {@link StackReferenceOutputDetails} object from it.
+     * The StackReferenceOutputDetails object has its value or secretValue set
+     * depending on whether the output is a secret or not.
+     */
+    public CompletableFuture<StackReferenceOutputDetails> outputDetailsAsync(String name) {
+        return Internal.of(this.output(name)).getDataAsync()
+                .thenApply((OutputData<?> data) -> {
+                    // Extraneous map call to turn the Optional<?> into an Optional<Object>.
+                    Optional<Object> value = data.getValueOptional().map(o -> o);
+                    StackReferenceOutputDetails.Builder builder = StackReferenceOutputDetails.builder();
+                    value.ifPresent(v -> {
+                        if (data.isSecret()) {
+                            builder.secretValue(v);
+                        } else {
+                            builder.value(v);
+                        }
+                    });
+                    return builder.build();
```

Comment:
I feel like there's a better way to construct this other than using a builder pattern since there's only a single parameter which is a value and there's a check to ensure that the caller didn't call `secretValue` and `value`.

An idea that I like better would be plumbing `data` into the `StackReferenceOutputDetails` constructor and having the logic inside `StackReferenceOutputDetails` set value or secretValue.

Reply:
Honestly, I don't love the builder thing here either.
We could do two package-local static constructors for StackReferenceOutputDetails, and use that instead:

```java
if (secret) {
  outputDetails = StackReferenceOutputDetails.fromSecret(..)
} else {
  outputDetails = StackReferenceOutputDetails.fromPlain(..)
}
```

Reply:
Although I would still want to keep the builder available for external users to be able to build valid instances of this object.

Reply:
> An idea that I like better would be plumbing data into the StackReferenceOutputDetails constructor and having the logic inside StackReferenceOutputDetails set value or secretValue.

Ah, I wrote the other reply too soon. Yeah, we could also do a static `fromOutput` constructor. The behavior in StackReferenceOutputDetails should be minimal, though since it's intended to be a plain data object.